### PR TITLE
Consume scalyr-agent-log also via fluentd

### PR DESCRIPTION
**Pull request by pheanex**
Signed-off-by: Konstantin Manna <konstantin.manna@zalando.de>

### CODE REVIEW

**Comment by mikkeloscar** [comment]
:+1:

**Comment by pheanex** [comment]
@christianberg @anton-ryzhov @vwiessner @astarnell please also review

File: cluster/manifests/logging-agent/config.yaml [hunk 1]
```diff
@@ -42,6 +55,18 @@ data:
       @type token
     </filter>
 
+    <filter scalyr-agent.**>
+      @type record_transformer
+      <record>
+        # s3_path is: ${application}/${component}/${namespace}/${environment}/${version}/${cont_name}
+        s3_path     "logging-agent/logging/visibility/{{ .Environment }}/v0.30/scalyr-agent"
+        application "logging-agent"
+        version     "v0.30"
```
Comment (anton-ryzhov) [inline cluster/manifests/logging-agent/config.yaml]: Is it possible to extract version from `deamonset.yaml`, or declare it in another place and use in this two?

File: cluster/manifests/logging-agent/daemonset.yaml [hunk 2]
```diff
@@ -244,4 +250,7 @@ spec:
       - name: fluentd-posfiles
         hostPath:
           path: /var/lib/fluentd
+
+      - name: scalyr-agent-logs
```
Comment (anton-ryzhov) [inline cluster/manifests/logging-agent/daemonset.yaml:254]: When this directory is [re]created? On pod restart? Or on container restart?

File: cluster/manifests/logging-agent/config.yaml [hunk 3]
```diff
@@ -42,6 +55,18 @@ data:
       @type token
     </filter>
 
+    <filter scalyr-agent.**>
+      @type record_transformer
+      <record>
+        # s3_path is: ${application}/${component}/${namespace}/${environment}/${version}/${cont_name}
+        s3_path     "logging-agent/logging/visibility/{{ .Environment }}/v0.30/scalyr-agent"
```
Comment (anton-ryzhov) [inline cluster/manifests/logging-agent/config.yaml]: `{{ .Environment }}` is not expanded

Comment (pheanex) [reply cluster/manifests/logging-agent/config.yaml]: As far as I understand this is something that is expanded when it is deployed/applied

File: cluster/manifests/logging-agent/daemonset.yaml (hunk 2 above)
Comment (pheanex) [reply cluster/manifests/logging-agent/daemonset.yaml:254]: https://kubernetes.io/docs/concepts/storage/volumes/#emptydir

File: cluster/manifests/logging-agent/config.yaml (hunk 3 above)
Comment (mikkeloscar) [reply cluster/manifests/logging-agent/config.yaml]: Yes, this is expanded by the [cluster-lifecycle-manager](https://github.com/zalando-incubator/cluster-lifecycle-manager) when applying the manifests.

File: cluster/manifests/logging-agent/config.yaml (hunk 1 above)
Comment (pheanex) [reply cluster/manifests/logging-agent/config.yaml]: We also hardcode it here https://github.com/zalando-incubator/kubernetes-on-aws/blob/dev/cluster/manifests/logging-agent/daemonset.yaml#L9
I do not know. Can we introduce some kind of variable in another manifest and reuse it here?
Do you know @mikkeloscar ?

Comment (anton-ryzhov) [reply cluster/manifests/logging-agent/config.yaml]: Hardcode in one place is declaration. In several places — unmaintainable thing

Comment (mikkeloscar) [reply cluster/manifests/logging-agent/config.yaml]: I can't think of any way to do this now.

Comment (mikkeloscar) [reply cluster/manifests/logging-agent/config.yaml]: I mean you could set the variable in `cluster/config-defaults.yaml` and use that, but I don't know how obvious it is when you need to change it.

Comment (aermakov-zalando) [reply cluster/manifests/logging-agent/config.yaml]: You could also move both resources into one yaml and then use template variables to define it once. Not sure if you think it'll be uglier or not. :)

**Comment by pheanex** [comment]
:+1:

**Review by anton-ryzhov — APPROVED** [review]

**Comment by anton-ryzhov** [comment]
:+1:

**Comment by christianberg** [comment]
Should we also use `logging-agent-version` variable here?
https://github.com/zalando-incubator/kubernetes-on-aws/blob/57990653d51d508f638f89ed41bfe045aec94456/cluster/manifests/logging-agent/daemonset.yaml#L86
Would need to republish the docker image with tag `v0.30` instead of `0.30`, though.
Even better would be to decouple the version of the `logging-agent` from the version of the `kubernetes-log-watcher`. (This can/should happen in a separate PR.)

**Comment by pheanex** [comment]
@christianberg IIRC we agreed if we do this, we will do it in another PR

**Comment by arjunrn** [comment]
:+1:

**Comment by jrake-revelant** [comment]
👍

**Pull request merged**